steam/autostart: ensmoothen Plymouth

K900 opened the pull request:
Before:

- boot starts
- Plymouth is shown
- Plymouth is shut down (due to greetd being ordered after plymouth-quit-wait)
- greetd starts
- session starts
- session takes over display

Now:

- boot starts
- Plymouth is shown
- greetd starts
- session starts
- Plymouth is shut down (because we tell it to)
- session takes over display

There's still a little bit of black screen in between, but it's a lot better.

K900 committed:
steam/autostart: ensmoothen Plymouth

ElvishJerricco commented on modules/steam/autostart.nix:
```diff
@@ -116,17 +116,15 @@ in
-      users.groups.jovian-greeter = {};
+      # We handle this ourselves in the greeter
+      systemd.services.plymouth-quit.enable = false;
+      systemd.services.plymouth-quit-wait.enable = false;
```
You shouldn't disable `plymouth-quit-wait.service`. That's there so that other things know when plymouth actually quit.

samueldr commented:
I'll rebase my plymouth futzing-about on top of this, and see how that goes.

K900 commented:
Nixpkgs side: https://github.com/NixOS/nixpkgs/pull/310786

K900 commented:
Merged the nixpkgs side, will merge this when channels update.

K900 committed:
flake.lock: Update

Flake lock file updates:

• Updated input 'nixpkgs':
    'github:NixOS/nixpkgs/7bb2ccd8cdc44c91edba16c48d2c8f331fb3d856' (2024-04-25)
  → 'github:NixOS/nixpkgs/33d1e753c82ffc557b4a585c77de43d4c922ebb5' (2024-05-15)

K900 committed:
manifest: ignore libversion

K900 committed:
hack: pick the first license for docs if it's a list
